### ISSUE TITLE
virtual-kubelet binary not available

### ISSUE THREAD

**Issue by bremnes** [virtual-kubelet/virtual-kubelet#324]
### Request for documentation improvement or releasing the binary
As I'm starting to research and try virtual-kubelet, I found this step to be non-intuitive. Several places the binary `virtual-kubelet` is being referenced, but there's no documentation on how to produce, download or install that binary. Perhaps I'm not seeing the obvious, but I am not able to find it.

Yes, there is a makefile and dockerfile, but those options are not ideal for everybody:
* To run a 'makefile' you have to have some software pre-installed. (For windows I installed make and go but got a lot of errors)
* The dockerfile allows you to spin up a working container with the virtual-kubelet binary available. Great, but now you have to set up the connection to the Kubernetes from scratch inside that container. Optionally copy the binary from that container out to where your kubernetes connection is set up, which you must if you for instance run kubernetes locally. Not ideal if you run on for instance Windows, where I assume you must have linux subsystem set up or similar to be able to execute that binary.

### Proposed solution
Ideally it should be possible for this project to release the pre-built binaries in the release section together with the source code. A lot of other projects do this where they ship pre-built versions of linux, osx and windows. See for instance istio at https://github.com/istio/istio/releases. Alternatively improve the documentation.

**Comment by robbiezhang** [comment]
The quickest way to try the VK is to use the helm chart we released in the repo.
https://github.com/virtual-kubelet/virtual-kubelet/tree/master/charts

Here is an instruction about how to use ACI in the AKS cluster:
https://github.com/virtual-kubelet/virtual-kubelet/blob/master/providers/azure/README.md#quick-set-up-with-the-aci-connector

I don't run VK binary on a Windows machine, unless running the tests by `go test ...`. The easiest way I can see is to install the Docker for Windows and runs the container image.

**Comment by bremnes** [comment]
Thanks for the feedback @robbiezhang. I noticed the charts, but they seemed kind of Azure oriented. Even though we are probably going to use it on AKS, we are testing against hyper.sh right now.

Based on the non-aks chart and [hyper.sh provider doc](https://github.com/virtual-kubelet/virtual-kubelet/tree/master/providers/hypersh) I added the environment variables `HYPER_ACCESS_KEY` and `HYPER_SECRET_KEY` to the deployment.yaml with the correct values and deployed the chart to kubernetes.

The container is stuck in CrashLoopBackOff status with the following log on two different Kubernetes clusters (see below):
```
time="2018-08-21T19:29:17Z" level=error msg="interface public not found"
/.kube/config
2018/08/21 19:29:17 Use AccessKey and SecretKey from HYPER_ACCESS_KEY and HYPER_SECRET_KEY
2018/08/21 19:29:17
 Host: tcp://us-west-1.hyper.sh:443
 AccessKey: E**********
 SecretKey: J**********
 InstanceType: s4
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1779b71]

goroutine 1 [running]:
github.com/virtual-kubelet/virtual-kubelet/vkubelet.(*Server).registerNode(0xc420038900, 0xc420038900, 0x0)
        /go/src/github.com/virtual-kubelet/virtual-kubelet/vkubelet/vkubelet.go:177 +0x5a1
github.com/virtual-kubelet/virtual-kubelet/vkubelet.New(0x7ffdfac9acc5, 0xf, 0x7ffdfac9acda, 0x5, 0x0, 0x0, 0xc42069b020, 0xd, 0x7ffdfac9aca7, 0x5, ...)
        /go/src/github.com/virtual-kubelet/virtual-kubelet/vkubelet/vkubelet.go:128 +0x4f3
github.com/virtual-kubelet/virtual-kubelet/cmd.glob..func1(0x2fc66e0, 0xc42053f500, 0x0, 0x8)
        /go/src/github.com/virtual-kubelet/virtual-kubelet/cmd/root.go:51 +0x14a
github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/spf13/cobra.(*Command).execute(0x2fc66e0, 0xc4200380a0, 0x8, 0x8, 0x2fc66e0, 0xc4200380a0)
        /go/src/github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/spf13/cobra/command.go:760 +0x2c1
github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x2fc66e0, 0x1, 0x0, 0x2fc6940)
        /go/src/github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/spf13/cobra/command.go:846 +0x30a
github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/spf13/cobra.(*Command).Execute(0x2fc66e0, 0xc42008e058, 0x0)
        /go/src/github.com/virtual-kubelet/virtual-kubelet/vendor/github.com/spf13/cobra/command.go:794 +0x2b
github.com/virtual-kubelet/virtual-kubelet/cmd.Execute()
        /go/src/github.com/virtual-kubelet/virtual-kubelet/cmd/root.go:62 +0x2d
main.main()
        /go/src/github.com/virtual-kubelet/virtual-kubelet/main.go:20 +0x20
```

From the [server log](https://github.com/virtual-kubelet/virtual-kubelet/tree/master/providers/hypersh#server-log) I would assume it's failing on the creation of a node in the cluster? 

Environment1: AKS; helm installed with RBAC
Environment2: My environment: Windows 10; docker edge which has a Kubernetes dev cluster included, set up in linux containers mode; helm installed with RBAC.

Any ideas? 
(This might be a separate issue, but perhaps not if I'm misunderstanding the documentation that exists.)

**Comment by robbiezhang** [comment]
Hmmm, looks like the hypersh provider is broken for a long while...
The exception is thrown from here:
https://github.com/virtual-kubelet/virtual-kubelet/blob/v0.5.1/vkubelet/vkubelet.go#L177

It didn't implement the NodeDaemonEndpoints function well:
https://github.com/virtual-kubelet/virtual-kubelet/blob/89921a08c1d5536db363c27ded55733764f1f221/providers/hypersh/hypersh.go#L445-L447

Actually, I don't think it's necessary as a function in the provider interface. Maybe, we should assign it in the vkubelet directly and remove the function.

**Comment by cberthiaume** [comment]
I'd like to second this request. I'm not a Go developer and it took some fumbling and frustration to get this built. Step 2 in [this|https://aws.amazon.com/blogs/opensource/aws-fargate-virtual-kubelet/] AWS document helped out quite a bit. It would much nicer if something like that was included in the documentation here.

The other problem I encountered was [this](https://stackoverflow.com/questions/42802278/install-context-package-in-golang) issue with Go versions lower than 1.7. I'm using Ubuntu 16.04 and it installed Go 1.6 by default. Knowing I needed to use 1.7 or above would have also helped quite a bit.
